homeassistant/components/tplink/light.py

```diff
@@ -212,27 +212,18 @@ def is_on(self):
         """Return True if device is on."""
         return self._light_state.state
 
-    def update(self):
+    async def async_update(self):
         """Update the TP-Link Bulb's state."""
         # State is currently being set, ignore.
         if self._is_setting_light_state:
             return
 
-        # Initial run, perform call blocking.
-        if not self._light_features:
-            self.do_update_retry(False)
-        # Subsequent runs should not block.
-        else:
-            self.hass.add_job(self.do_update_retry, True)
-
-    def do_update_retry(self, update_state: bool) -> None:
-        """Update state data with retry.""" ""
         try:
             # Update light features only once.
             self._light_features = (
-                self._light_features or self.get_light_features_retry()
+                self._light_features or self.get_light_features()
```

Comment:
If `self._light_features` is 0, this is executed each time. Probably better check for `None`.

Reply:
I think this should never come back as 0, as all the bulbs support one of those features.